circuit-lib/circuit-lib.circom/benchmarks/light/transaction_light.ts

```diff
@@ -0,0 +1,169 @@
+const bench = require("micro-bmark");
+const snarkjs = require("snarkjs");
+const buildPoseidonOpt = require("circomlibjs").buildPoseidonOpt;
+
+import { Keypair as SolanaKeypair, PublicKey } from "@solana/web3.js";
+import { BN } from "@coral-xyz/anchor";
+
+import {
+  Account,
+  Utxo,
+  FEE_ASSET,
+  Provider as LightProvider,
+  MINT,
+  Transaction,
+  TransactionParameters,
+  Action,
+  IDL_VERIFIER_PROGRAM_ZERO,
+  IDL_VERIFIER_PROGRAM_TWO,
+} from "@lightprotocol/zk.js";
+
+import { bs58 } from "@coral-xyz/anchor/dist/cjs/utils/bytes";
+import { MerkleTree } from "../../../circuit-lib.js/lib/index.js";
+import { stringifyBigInts } from "maci-crypto";
+
+// This will extract the type of the constructor parameters of Transaction CLass
+type TransactionInputs = ConstructorParameters<typeof Transaction>[0];
+
+function addPrivateKey(transaction: Transaction, account: Account) {
+  transaction.proofInput["inPrivateKey"] = transaction.params.inputUtxos.map(
+    (utxo: Utxo) => {
+      if (utxo.publicKey == account.pubkey) {
+        return account.privkey;
```

Comment:
This is soon not going to work anymore because we will make account.privkey private.
Please find a different solution.
For example you could do a mock key pair which uses a random 31 bytes number as private key and the poseidon hash of it as pubkey.

Reply:
i'll generate the same keypair by the static methods:
```typescript
  const account_privateKey = Account.generateShieldedPrivateKey(seed32, poseidon);
  const account_publicKey = Account.generateShieldedPublicKey(account_privateKey, poseidon);
```
but if you want `privKey` to be private then you also need to make `seed` in the account class private as well.